### ISSUE TITLE
Better Integration with Dash to Dock? or Regular Dash?

### ISSUE THREAD

**Issue by ghost** [mlutfy/hidetopbar#149]
At the moment my setup [looks like this](https://user-images.githubusercontent.com/29314727/32437367-01991a88-c2df-11e7-9d1b-ea3d8fc2445f.png).

Is there anyway to force the Dash to adapt so the top bar doesn't overwrite it when only the desktop is showing? (in this case I am using the Dash-to-Dock extension).

**Comment by gfiorav** [comment]
Having the same issue. I've attempted a workaround by changing Ubuntu Dock/Dash to Dock CSS to add top padding, but I've desisted in favor of using the dock in bottom position. Watching this issue for further updates.

Thanks!

**Comment by LauBrust** [comment]
Same issue on a fresh ubuntu 18.04.
I've solved it deactivating `extend-height` option with `dconf` at `org/gnome/shell/extensions/dash-to-dock`
If you have many apps then it does not work neither.

**Comment by gitfib** [comment]
Workaround: add this to the end of extension.js
`(function(getWorkAreaForMonitor) {
  imports.ui.layout.LayoutManager.prototype.getWorkAreaForMonitor = function (monitorIndex) {
    let workarea = getWorkAreaForMonitor.call(this, monitorIndex);
    workarea.y = this.panelBox.height;
    return workarea;
  };
}(imports.ui.layout.LayoutManager.prototype.getWorkAreaForMonitor));
`

**Comment by pabloz** [comment]
To me, the above workaround is pushing down the "Show Applications", so I added this to the dash-to-dock app, in `~/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/stylesheet.css`:

```css
/* fix to prevent the topbar overlaps the first icon in the docker */
#dashtodockContainer.left .dash-item-container:first-child > StButton:first-child,
#dashtodockContainer.right .dash-item-container:first-child > StButton:first-child {
   margin-top: 30px;
}
```
Not great as a workaround as it'll probably disappear on the next update of the extension (plus, assumes you have dash-to-dock installed), but it does the job for now.

**Comment by tuxor1337** [comment]
@TIIUNDER mentioned something about a workaround here: https://github.com/mlutfy/hidetopbar/issues/243#issuecomment-658399480

**Comment by TIIUNDER** [comment]
Yes, you just need to disable and re-enable the Hide Top Bar extension. This way after enabling the top bar overlaps the dash/dock again. To achieve this automatically I created the following simple bash script:

```
#!/bin/sh

gnome-extensions disable hidetopbar@mathieu.bidon.ca
gnome-extensions enable hidetopbar@mathieu.bidon.ca
```

Put this into a file, for example `~/bin/toggle_topbar.sh`, make it runnable using `chmod +x` and put this script into your start programs. This way it gets automatically launched on login.

![Startprogramme_2020-07-15_11:31:06](https://user-images.githubusercontent.com/1225378/87529123-d20dd900-c68e-11ea-847f-c573587e8fd1.png)

![Startprogramm bearbeiten_2020-07-15_11:31:34](https://user-images.githubusercontent.com/1225378/87529142-d639f680-c68e-11ea-96ec-600ca18c1834.png)

**Comment by tuxor1337** [comment]
Just wanted to drop here that (since we merged #322), this is now an issue on the side of Dash to Dock. They could solve this issue by making use of the functionality implemented in #322.
